pkg/apis/internal.admin.acorn.io/v1/default.go

```diff
@@ -18,24 +21,33 @@ func getCurrentClusterComputeClassDefault(ctx context.Context, c client.Client)
 		return clusterComputeClasses.Items[i].Name < clusterComputeClasses.Items[j].Name
 	})
 
-	var defaultCCC *ClusterComputeClassInstance
+	var defaultCCC, projectSpecifiedCCC *ClusterComputeClassInstance
 	for _, clusterComputeClass := range clusterComputeClasses.Items {
 		if clusterComputeClass.Default {
 			if defaultCCC != nil {
 				return nil, fmt.Errorf(
 					"cannot establish defaults because two default computeclasses exist: %v and %v",
 					defaultCCC.Name, clusterComputeClass.Name)
 			}
-			t := clusterComputeClass // Create a new variable that isn't being iterated on to get a pointer
-			defaultCCC = &t
+
+			// Create a new variable that isn't being iterated on to get a pointer
+			defaultCCC = z.Pointer(clusterComputeClass)
+		}
+
+		if projectSpecified != "" && clusterComputeClass.Name == projectSpecified {
+			projectSpecifiedCCC = z.Pointer(clusterComputeClass)
 		}
 	}
 
+	if projectSpecifiedCCC != nil {
+		return projectSpecifiedCCC, nil
+	}
```

Comment:
Spoke about this offline: if a user specifies a compute class that doesn't exist, then we should error here.

Additionally, we should check that the default region on the project is supported by the compute class.

Lastly, validation should be added for the user changing the default region on the project to ensure that the default compute class, if one is set, supports the region.